ember-toucan-core/src/components/form/file-input-field.ts

```diff
@@ -0,0 +1,85 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+// Note, unlike other inputs the first parameter is not @value but @files
+export type OnFileUploadChangeCallback = (
+  files: File[],
+  e: Event | InputEvent
+) => void;
+
+interface ToucanFormFileInputFieldComponentSignature {
+  Element: HTMLInputElement;
+  Args: {
+    accept?: string;
+    error?: string;
+    files?: File[];
+    hint?: string;
+    label: string;
+    isDisabled?: boolean;
+    multiple?: boolean;
+    onChange?: OnFileUploadChangeCallback;
+    onDeleteFile?: (file: File) => void;
```

Comment:
I wonder about the API design of these two callbacks here.
* when a file is deleted, a user could maybe expect `@onChange` to be called also, as the file list _has_ changed. But this does not seem to happen, right?
* we could also have just `@onChange`, and call that with the now smaller list of files when deleting a file
* or we rename `@onChange` to `@onAddFile` or something like this, to make this clear, when we want to have different change callbacks

🤔